service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,70 @@
         </actions>
     </service>
 
+    <service verb="get" noun="AutoPackageInfoList">
+        <implements service="mantle.shipment.CarrierServices.get#AutoPackageInfo"/>
+        <actions>
+            <iterate list="itemInfoList" entry="itemInfoMap">
+                <entity-find entity-name="mantle.shipment.ShipmentItemSource" list="shipmentSourceList">
+                    <econdition field-name="orderId" from="itemInfoMap.orderId"/>
+                </entity-find>
+                <iterate list="shipmentSourceList" entry="shipmentSourceMap">
+                    <set field="shipmentId" from="shipmentSourceMap.shipmentId"/>
+                </iterate>
+            </iterate>
+            <entity-find entity-name="mantle.shipment.ShipmentItemSource" list="itemSourceList">
+                <econdition field-name="shipmentId"/>
+                <order-by field-name="shipmentPackageSeqId"/>
+            </entity-find>
+            <set field="packageInfoList" from="[]"/>
+            <entity-find entity-name="mantle.shipment.ShipmentPackage" list="packageList">
+                <econdition field-name="shipmentId"/>
+                <order-by field-name="shipmentPackageSeqId"/>
+            </entity-find>
+            <entity-find entity-name="mantle.shipment.ShipmentPackageContent" list="shipmentPackageContentList">
+                <econdition field-name="shipmentId"/>
+                <order-by field-name="productId"/>
+            </entity-find>
+            <iterate list="packageList" entry="pkg">
+                <filter-map-list list="shipmentPackageContentList" to-list="curContentList">
+                    <field-map field-name="shipmentPackageSeqId" from="pkg.shipmentPackageSeqId"/>
+                </filter-map-list>
+                <set field="quantityByProductId" from="[:]"/>
+                <iterate list="curContentList" entry="curContent">
+                    <script>quantityByProductId.put(curContent.productId, curContent.quantity)</script>
+                </iterate>
+
+                <script>packageInfoList.add([shipmentBoxTypeId:pkg.shipmentBoxTypeId, weight:pkg.weight,
+                    weightUomId:pkg.weightUomId, shipmentPackageSeqId:pkg.shipmentPackageSeqId,
+                    quantityByProductId:quantityByProductId])
+                </script>
+            </iterate>
+        </actions>
+    </service>
+
+    <service verb="get" noun="ItemInfoList">
```

Comment:
This service is added only for testing I believe. Please remove. The autoPackageInfo service will be called in calculate#OrderPartShipping if service is configured in the shipping gateway record. Please check.